services/meta/client.go

```diff
@@ -539,7 +539,23 @@ func (c *Client) Authenticate(username, password string) (*UserInfo, error) {
 	userInfo := c.cacheData.User(username)
 	c.mu.RUnlock()
 	if userInfo == nil {
-		return nil, ErrUserNotFound
+		// Instead of instantly returning when the username is not found, compare against a valid bcrypt hash
+		// so that we don't leak whether the username was valid via how quickly the server responds.
+		// See https://cwe.mitre.org/data/definitions/208.html for more details.
+
+		// Create fake bcrypt hash with Version=2, Cost=<current cost setting>, Salt+Hash=<doesn't really matter>.
+		fakeHash := fmt.Sprintf("$2$%02d$00000000000000000000000000000000000000000000000000000", bcryptCost)
```

Comment:
The difficulty of doing this once for the package is that `bcryptCost` is reassigned during in the package tests, so the benchmarks become skewed if this path uses an expensive bcrypt call and all the other paths use bcrypt.MinCost.

It was unclear to me whether `bcrypt.GenerateFromPassword` had the same cost as `bcrypt.CompareHashAndPassword`. I probably should have benchmarked it.